optimum_benchmark/benchmarks/training.py

```diff
@@ -142,6 +145,31 @@ class TrainingConfig(BenchmarkConfig):
         }
     )
 
+    # These configurations are specific to PyTorch DDP.
+    # Copied from https://github.com/pytorch/pytorch/blob/v2.0.0/torch/distributed/launcher/api.py#L29, adjusting to the defaults of torch.distributed.run
+    use_ddp: bool = False
+    ddp_config: DictConfig = DictConfig(
```

Comment:
yes because it shouldn't be backend-specific. it should only contain what's necessary and backend independent. other things can be added by the user through the command line or yaml config when needed.

the idea of having training specific config here is to allow easy benchmarking of multiple backends with one command line `-m backend=pytorch,neural-compressor,tensorflow,flax benchmark=training` for simple configs (I wanna choose the best ddp backend for example without going the rabbit hole of parameters optimization). and to also allow very specific settings for one particular backned `backend=pytorch benchmark=training benchmark.ddp_config.pytorch_specific_param=foo` but in that case the user should know what they're doing.
if you wanna have default values for a pytorch-specific params you can define it in `pytorch_specific_param = ddp_config.get("pytorch_specific_param", default_value)`.

Reply:
It's the same as `training_arguments` where only the common most necessary params are defined for the same reason mentioned above (one command line to sweep over ort and pytorch training -> quick decision about which one to use).
but it doesn't stop someone who knows the orttrainer api to add whatever params they wish to add for an ort training specific setting.

Reply:
`ddp_config` is specific to PyTorch until tested otherwise. I don't expect all backends to support DDP either, nor do I expect all backends to support and rely on transformers TrainingArguments. It is not an assumption we should make.

The point I am making is that adding the `ddp_config` here makes little sense as it is not scalable in the sense that it is backend-specific, and mixing up backend-specific config with training-specific config is IMO a no-go.

Note that the defaults from `torch.distributed.run` CLI are different from the dataclass `LaunchConfig`, as I noted in a comment. As the goal is to reproduce the behavior of `torch.distributed.run`, I included its defaults here. Do you think there is a better way?

Reply:
with these two propositions, the output config is bloated in all cases. if I launch a pytorch dp training I'll always have ddp_config there whether in backend or benchmark.
So the straightforward solution for me that solves most problems while also allowing
- `-m backend=pytorch,neural-compressor,tensorflow,flax benchmark=training`
- keeps most of the backend logic in the backend.
- keeps benchmarking logic in the benchmark.
is to add a benchmark named training-ddp or training-pytorch-ddp for example that handles this edge case. the choice is yours. if we're gonna add a benchmark by backend at least let's only do it when forced by that backend's ddp and not pytorch's.

Reply:
With the solution you propose (which is what I originally implemented with the `PyTorchTrainingBenchmark` subclass) the `training-pytorch-ddp` could not be used with -m backend=pytorch,neural-compressor,tensorflow,flax benchmark=training. Is that an issue?

> with these two propositions, the output config is bloated in all cases. if I launch a pytorch dp training I'll always have ddp_config there whether in backend or benchmark.

Yes I think I can fix that.

Reply:
that would still be possible if `training-pytorch-ddp` only handles the edge case, and dp training is kept for `training` benchmark. my understanding from the previous implementation is that all backends need a custom benchmark to handle their training (dp and ddp).

Reply:
What I mean is that you could not use PyTorch DDP with `-m backend=pytorch,neural-compressor,tensorflow,flax benchmark=training` in the solution you propose.

It seems to be about finding a compromise here - any is fine to me.